mod/storage/pkg/deposit/store.go

```diff
@@ -37,12 +39,18 @@ const KeyDepositPrefix = "deposit"
 // the deposit indexes are tracked outside of the kv store.
 type KVStore[DepositT Deposit[DepositT]] struct {
 	store sdkcollections.Map[uint64, DepositT]
-	mu    sync.RWMutex
+
+	// mu protects store for concurrent access
+	mu sync.RWMutex
+
+	// logger is used for logging information and errors.
+	logger log.Logger
```

Comment:
not sure I want to keep logger here, or at least I should move `Info`s to `Debug`s. This is because this store is called by both block builder and verified, so we want to avoid double logging, or at least mark the client